mirrord-layer/src/go/hooks.rs

```diff
@@ -0,0 +1,423 @@
+        };
+        debug!(
+            "c_abi_syscall_handler >> result -> {res:?}, errorno -> {:?}",
+            errno().0
+        );
+        match res {
+            -1 => -errno().0 as i64,
+            _ => res as i64,
+        }
+    }
+
+    /// [Naked function] Syscall & Syscall6 handler - supports upto 6 params, mainly used for
+    /// accept4 Note: Depending on success/failure Syscall may or may not call this handler
+    #[no_mangle]
+    unsafe extern "C" fn c_abi_syscall6_handler(
+        syscall: i64,
+        param1: i64,
+        param2: i64,
+        param3: i64,
+        param4: i64,
+        param5: i64,
+        param6: i64,
+    ) -> i64 {
+        debug!("C ABI handler received `Syscall6 - {:?}` with args >> arg1 -> {:?}, arg2 -> {:?}, arg3 -> {:?}, arg4 -> {:?}, arg5 -> {:?}, arg6 -> {:?}", syscall, param1, param2, param3, param4, param5, param6);
+        let res = match syscall {
+            libc::SYS_accept4 => {
+                accept4_detour(param1 as _, param2 as _, param3 as _, param4 as _) as i64
+            }
+            _ => syscall_6(syscall, param1, param2, param3, param4, param5, param6),
+        };
+        debug!(
+            "c_abi_syscall_handler >> result -> {res:?}, errorno -> {:?}",
+            errno().0
+        );
+        match res {
+            -1 => -errno().0 as i64,
+            _ => res as i64,
+        }
+    }
+
+    /// [Naked function] 3 param version (Syscall6) for making the syscall, libc's syscall is not
+    /// used here as it doesn't return the value that go expects (it does translation)
+    #[naked]
+    unsafe extern "C" fn syscall_3(syscall: i64, arg1: i64, arg2: i64, arg3: i64) -> i64 {
+        asm!(
+            "mov    rax, rdi",
+            "mov    rdi, rsi",
+            "mov    rsi, rdx",
+            "mov    rdx, rcx",
+            "syscall",
+            "ret",
+            options(noreturn)
+        )
+    }
+
+    /// [Naked function] 6 param version, used by Rawsyscall & Syscall
+    #[naked]
+    unsafe extern "C" fn syscall_6(
+        syscall: i64,
+        arg1: i64,
+        arg2: i64,
+        arg3: i64,
+        arg4: i64,
+        arg5: i64,
+        arg6: i64,
+    ) -> i64 {
+        asm!(
+            "mov    rax, rdi",
+            "mov    rdi, rsi",
+            "mov    rsi, rdx",
+            "mov    rdx, rcx",
+            "mov    r10, r8",
+            "mov    r8, r9",
+            "mov    r9, QWORD PTR[rsp]",
+            "syscall",
+            "ret",
+            options(noreturn)
+        )
+    }
+
+    /// Note: We only hook "RawSyscall", "Syscall6", and "Syscall" because for our usecase,
+    /// when testing with "Gin", only these symbols were used to make syscalls.
+    /// Refer:
+    ///   - File zsyscall_linux_amd64.go generated using mksyscall.pl.
+    ///   - https://cs.opensource.google/go/go/+/refs/tags/go1.18.5:src/syscall/syscall_unix.go
+    pub(crate) fn enable_socket_hooks(interceptor: &mut Interceptor, binary: &str) {
```

Comment:
I'm okay with not doing file hooks right now, but 99% we need to do similar for file, so please open an issue so we won't forget it :)